ushahidi/platform-client-mzima
# [Outreachy Task Submission] Fix Missing Search Button Value Text

# Introduction
The search form component in both the `Data` and `Map` view has a problem. The **search** button has only an icon and has no value text. While this is just a stylistic choice, it raises an issue with any accessibility checking tool.  It does this because a button having no value text is a violation of WCAG Success Criterion 1.1.1 (Level A) - Non-text content. [See](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html). See a screenshot of the pertinent component:

![image](https://github.com/ushahidi/platform-client-mzima/assets/29470516/d98884fc-8a00-43c4-9a8a-4b0e7116c95c)
> ***Note the search icon***

# How to Test the PR
## Manually Testing
1. Open your browser and log into your running Ushahidi deployment (from localhost).
2. Navigate to either the `data` or `map` section.
3. Try right-clicking on the search form at the top of the screen and selecting 'inspect'.
4. View the properties, and look for a `span` element with a class of `mzima-button__text`. See screenshot below:
![image](https://github.com/ushahidi/platform-client-mzima/assets/29470516/31a4bd15-d8c3-40b7-9fca-108a0b91fc56)
5. You will not see any text in it.
6. Stop the running Ushahidi deployment in your terminal.
7. Checkout to this PR branch.
8. Re-run your Ushahidi deployment.
10. Repeat steps 1-4.
11. You will see a value in the `span` element this time around, and this text will be visually hidden.

## Using an Accessibility Checker
If you use a tool like [WAVE evaluation tool](https://chromewebstore.google.com/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh), it's as simple as following steps 1 and 2 above, and then turning on the browser extension. It will flag the issue. Checkout to the PR and do this again, you will notice the issue is no longer flagged. 

## Dependency
It has a dependency on a previous [PR](https://github.com/ushahidi/platform-client-mzima/pull/860)

## Fixes
It partially fixes [#4805](https://github.com/ushahidi/platform/issues/4805)
